### PR TITLE
🐛 Multi-output nodes

### PR DESCRIPTION
**Pull request by simon1hofmann**
## Description

Fixes a corner case in `ortho` where a gate has multiple output nodes connected to it.
Thanks to @hibenj for pointing out this solution in his implementation of `orthogonal_ordering_network`.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [ ] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by hibenj** [comment]
The code seems to be working. Before merging you might test the following: I guess you already checked for all benchmarks, that this is working. But you could also check, if the number of added rows corresponds to the number of multi PO nodes. But since you first determine the layout size and then insert the PO nodes, everything should be working perfectly. I just know i had issues with this before.

**Comment by simon1hofmann** [comment]
> The code seems to be working. Before merging you might test the following: I guess you already checked for all benchmarks, that this is working. But you could also check, if the number of added rows corresponds to the number of multi PO nodes. But since you first determine the layout size and then insert the PO nodes, everything should be working perfectly. I just know i had issues with this before.

Additional rows are added based on the number of multi-output nodes in `determine_layout_size()`:

`// for multi-output nodes, add another row
    y += num_multi_output_pos;`

**Review by hibenj — COMMENTED** [review]
The code seems to be working. But since you already iterate through all pos at the beginning of orthogonal_impl adn therefore compute all multi_out_nodes, you might be able to use them in the main loop of ortho to place the multi_out_nodes. With this implementation you need to iterate through all pos twice.
So i approved it, since its working, but i guess its up to @marcelwa if this implementation makes sense.

**Review by hibenj — DISMISSED** [review]

**Comment by simon1hofmann** [comment]
> The code seems to be working. But since you already iterate through all pos at the beginning of orthogonal_impl and therefore compute all multi_out_nodes, you might be able to use them in the main loop of ortho to place the multi_out_nodes. With this implementation you need to iterate through all pos twice. So i approved it, since its working, but i guess its up to @marcelwa if this implementation makes sense.

The problem ist that multi-output POs usually appear in a different order in `foreach_po` compared to `foreach_node`.
I now added a check for the appearance of multi-output POs in the network and only if they are present, the POs are placed after the main loop. Otherwise, `ortho` places all POs in the main loop.

**Comment by simon1hofmann** [comment]
> > The code seems to be working. But since you already iterate through all pos at the beginning of orthogonal_impl and therefore compute all multi_out_nodes, you might be able to use them in the main loop of ortho to place the multi_out_nodes. With this implementation you need to iterate through all pos twice. So i approved it, since its working, but i guess its up to @marcelwa if this implementation makes sense.
> 
> The problem ist that multi-output POs usually appear in a different order in `foreach_po` compared to `foreach_node`. I now added a check for the appearance of multi-output POs in the network and only if they are present, the POs are placed after the main loop. Otherwise, `ortho` places all POs in the main loop.

Actually, this can also happen for networks without mulit-ouput POs, which explains why the layout for `clpl` was not equivalent. I added another check to test that.

File: include/fiction/algorithms/physical_design/orthogonal.hpp
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/orthogonal.hpp:652]: Instead of manually incrementing x and y positions, rather use `east()` and `south()`

File: include/fiction/algorithms/physical_design/orthogonal.hpp
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/orthogonal.hpp:639]: here too

File: include/fiction/algorithms/physical_design/orthogonal.hpp
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/orthogonal.hpp:657]: here too

**Review by marcelwa — CHANGES_REQUESTED** [review]

**Comment by simon1hofmann** [comment]
@marcelwa I refactored the code to always place POs at the end, as this will solve both problems simultaneously.

File: include/fiction/algorithms/physical_design/orthogonal.hpp
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/orthogonal.hpp:408]: If `node2pos` is not updated, it may be `const`

File: include/fiction/algorithms/physical_design/orthogonal.hpp
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/orthogonal.hpp:477]:
```suggestion
        // find multi-output nodes
```

**Review by marcelwa — CHANGES_REQUESTED** [review]
Just two tiny comments, then it is ready to be merged

**Pull request merged**
